### PR TITLE
[SPARK-49748][CORE][FOLLOWUP] Add `getCondition` and deprecate `getErrorClass` in `QueryCompilationErrorsSuite`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr is following up https://github.com/apache/spark/pull/48196

### Why are the changes needed?
Revise the logic of the newly added logic.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala [hunk 1]
```diff
@@ -1003,7 +1003,7 @@ class QueryCompilationErrorsSuite
         val exception = intercept[AnalysisException] {
           sql(queryWithTrailingComma)
         }
-        assert(exception.getErrorClass === "TRAILING_COMMA_IN_SELECT")
```
Comment (panbingkun) [inline sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala:1006]: The newly added logic comes from https://github.com/apache/spark/pull/48370/files#diff-1fd98ab15bfc9b8068b37cc5cfbc3376ee63ca2d18600a502db3012d24915424R1006

**Comment by panbingkun** [comment]
I'm not sure if there will be similar uses in the future, because in our code, it's not mandatory not to use `getErrorClass`.

**Comment by panbingkun** [comment]
cc @MaxGekk

**Comment by MaxGekk** [comment]
> I'm not sure if there will be similar uses in the future, because in our code, it's not mandatory not to use getErrorClass.

@panbingkun Feel free to add a rule for style checker, and ban `getErrorClass`. Just in case, it should be a context-aware, so, it shouldn't ban the `getErrorClass` calls on non-`SparkThrowable`.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

**Pull request closed without merging**